src/main/java/net/sf/jabref/logic/importer/fileformat/MrDLibImporter.java

```diff
@@ -0,0 +1,275 @@
+/**
+ *
+ */
+package net.sf.jabref.logic.importer.fileformat;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import net.sf.jabref.logic.importer.Importer;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.util.FileExtensions;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ *
+ *
+ */
+public class MrDLibImporter extends Importer {
+
+    private final String NAME = "MrDLibImporter";
+    private final String DESCRIPTION = "Takes valid xml documents. Parses from MrDLib API a BibEntriy";
+    private static final Log LOGGER = LogFactory.getLog(MrDLibImporter.class);
+
+
+    /**
+     *
+     * @see net.sf.jabref.logic.importer.Importer#isRecognizedFormat(java.io.BufferedReader)
+     */
+    @Override
+    public boolean isRecognizedFormat(BufferedReader input) throws IOException {
+        String recommendationsAsString = BufferedReaderToParsableString(input);
+        // check for valid format
+        try {
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            SAXParser saxParser = factory.newSAXParser();
+            DefaultHandler handler = new DefaultHandler() {
+                // No Processing here. Just check for valid xml.
+            };
+
+            try {
+                InputStream stream = new ByteArrayInputStream(recommendationsAsString.toString().getBytes());
+                saxParser.parse(stream, handler);
+            } catch (Exception e) {
+                LOGGER.error(e.getMessage(), e);
+                return false;
+            }
+        } catch (ParserConfigurationException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        } catch (SAXException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        }
+        return true;
```

Comment:
@Siedlerchr If I have the parsing model of SAX correctly in mind, no exception means that parsing was okay. So if the execution makes it until this line, everything should be fine.